services/models/unit.py

```diff
@@ -267,24 +270,37 @@ def highlight_names(self):
         )
 
     @classmethod
-    def get_search_column_indexing(cls):
+    def get_search_column_indexing(cls, lang):
         """
         Defines the columns to be indexed to the search_column
         ,config language and weight.
         """
-        return [
-            ("name_fi", "finnish", "A"),
-            ("name_sv", "swedish", "A"),
-            ("name_en", "english", "A"),
-            ("service_names_fi", "finnish", "B"),
-            ("service_names_sv", "swedish", "B"),
-            ("service_names_en", "english", "B"),
-            ("extra", None, "C"),
-            ("address_zip", None, "D"),            
-            ("street_address_fi", "finnish", "D"),
-            ("street_address_sv", "swedish", "D"),
-            ("street_address_en", "english", "D"),     
-        ]
+        if lang == "fi":
+            return [
+                ("name_fi", "finnish", "A"),
+                ("service_names_fi", "finnish", "B"),
+                ("extra", None, "C"),
+                ("address_zip", None, "D"),
+                ("street_address_fi", "finnish", "D"),
+            ]
+        elif lang == "sv":
+            return [
+                ("name_sv", "swedish", "A"),
+                ("service_names_sv", "swedish", "B"),
+                ("extra", None, "C"),
+                ("address_zip", None, "D"),
+                ("service_names_sv", "swedish", "B"),
+            ]
+        elif lang == "en":
+            return [
+                ("name_en", "english", "A"),
+                ("service_names_en", "english", "B"),
+                ("extra", None, "C"),
+                ("address_zip", None, "D"),
+                ("service_names_en", "english", "B"),
```

Comment:
This should be most probably fixed from `service_names_en ` to `street_address_en`

Reply:
Yes, fixed